proc.c

```diff
@@ -628,3 +631,59 @@ procdump(void)
       cprintf("\n");
   }
 }
+
+
+int kthread_create(void (*start_func)(), void* stack){
+    //TODO
+    return 0;
+}
+
+int kthread_id(){
+    return my_thread()->tid;
+}
+
+void kthread_exit(){
+    //struct proc *curproc = myproc();
+    struct thread * curr_thread = my_thread();
+    boolean lastthread = TRUE;
+    struct proc* proc = myproc();
+    acquire(&proc->proclock);
+    struct thread * t;
+    for (t = proc->threads; t < &proc->threads[NTHREADS]; t++){
+        if(t->t_state == T_RUNNABLE | t->t_state==T_RUNNING |t->t_state == T_SLEEPING) {
+            lastthread = FALSE;
+            break;
+        }
+
+    }
+    if(lastthread){
+        exit();
+    }else {
+        curr_thread->t_state = T_TERMINATED;
```

Comment:
also add a call to
```
wakeup(proc);
```
in order to wake up all threads who called **kthread_join** in order to notify them about the exit.
and note that wakeup first acquire ptable lock, so you must release **proc->proclock** before to keep the right locking order to avoid deadlocks (ptable must be acquired before proc->proclock)